### PR TITLE
Support input plugin systemd format

### PR DESCRIPTION
**Pull request by jhsiaosuse**
Previously, this plugin cannot work correctly if the input plugin is [systemd](https://docs.fluentbit.io/manual/input/systemd) because the converter does not handle systemd key.
Add the following keys to supports systemd:
-  MESSAGE
- _HOSTNAME
- _COMM
- SYSLOG_IDENTIFIER
- _PID
- SYSLOG_FACILITY
- PRIORITY

Also, updates the base ubuntu image from `xenial` to `bionic`.

### CODE REVIEW

**Comment by hev** [comment]
@jhsiaosuse thanks for the PR. I think we'll want to discuss this one as a team as it breaks some current conventions we have around how we use the 5424 spec. A few of our team members are out at Kubecon so apologies in advance for any slowness.

**Comment by jhsiaosuse** [comment]
> @jhsiaosuse thanks for the PR. I think we'll want to discuss this one as a team as it breaks some current conventions we have around how we use the 5424 spec. A few of our team members are out at Kubecon so apologies in advance for any slowness.

:ok:, waits for your reply

File: pkg/syslog/out.go [hunk 1]
```diff
@@ -390,26 +441,27 @@ func convert(
...
+			rfc5424Msg.Hostname = vmID
+		}
 
-	if host == "" {
-		host = vmID
+		// Add to structured data.
+		for _, label := range labelParams {
+			rfc5424Msg.AddDatum("kubernetes", label.Name, label.Value)
+		}
+		rfc5424Msg.AddDatum("kubernetes", "namespace_name", nsName)
```
Comment (wfernandes) [inline pkg/syslog/out.go]: This ID seems incorrect - It was `kubernetes@47450` now it is just `kubernetes`. But I do like the use of `AddDatum` function. Maybe we can refactor our stuff to use that if possible.

File: Dockerfile [hunk 2]
```diff
@@ -66,11 +66,11 @@ COPY /config/fluent-bit.conf \
      /config/parsers_cinder.conf \
      /fluent-bit/etc/
 
-FROM ubuntu:xenial
+FROM ubuntu:bionic
 
 RUN apt-get update \
     && apt-get dist-upgrade -y \
-    && apt-get install --no-install-recommends ca-certificates libssl1.0.2 -y libsasl2-2 \
+    && apt-get install --no-install-recommends ca-certificates libssl1.0.0 -y libsasl2-2 systemd \
```
[the remark this answers is not in the document]
Comment (jhsiaosuse) [reply Dockerfile]: Thanks.

File: Dockerfile [hunk 3]
```diff
@@ -16,7 +16,7 @@ RUN go build \
     -mod=vendor \
     cmd/main.go
 
-FROM ubuntu:xenial as builder
+FROM ubuntu:bionic as builder
```
[the remark this answers is not in the document]
Comment (jhsiaosuse) [reply Dockerfile]: I encounter an issue that I cannot start fluent-bit successfully in `ubuntu:xenial`, very strange, I will keep digging out.

File: pkg/syslog/out.go (hunk 1 above)
Comment (jhsiaosuse) [reply pkg/syslog/out.go]: What this ID exactly means? I check RFC5424 spec but cannot found it.

**Comment by hev** [comment]
@jhsiaosuse - I am going to go ahead and close this PR. While I see the value in allowing for tagging systemd, all of our applications of this plugins (Knative, PKS, etc) are for workloads running within cluster and we'd like to keep this plugin focused on that. This would make more sense as a separate plugin in my mind.

**Pull request closed without merging**